### ISSUE TITLE
[API Proposal]: Add architecture ARMv6 to Architecture enum

### ISSUE THREAD

**Issue by directhex** [dotnet/runtime#63310]
### Background and motivation

All architectures, whether officially Microsoft supported or not, need to exist within the System.Runtime.InteropServices.Architecture enum. We want to introduce a new community-supported architecture, `armv6`, which needs to be distinct from `arm` to avoid compatibility confusion (i.e. we want to be certain NuGet will never try and obtain unmanaged code for `arm` on a platform running `armv6`, or the app may throw SIGILL).

Within .NET, `arm` means one specific subset of the potential universe of ARM instruction sets (ARMv7 base instructions, VFPv3 floating point support and registers, hard-float ABI, Thumb2 compact instruction set), we need another architecture value for a different subset - in this case, the compile target for early models and smaller models of Raspberry Pi single board computer (ARMv6 base instructions, VFPv2 floating point support and registers, hard-float ABI, no compact instruction set)

Ref: https://github.com/dotnet/runtime/issues/7764
Ref: https://github.com/dotnet/runtime/pull/62594

### API Proposal

```C#
diff --git a/src/libraries/System.Runtime/ref/System.Runtime.cs b/src/libraries/System.Runtime/ref/System.Runtime.cs
index 4036cc486b1..46a99b8d175 100644
--- a/src/libraries/System.Runtime/ref/System.Runtime.cs
+++ b/src/libraries/System.Runtime/ref/System.Runtime.cs
@@ -13431,6 +13431,7 @@ public enum Architecture
         Arm64 = 3,
         Wasm = 4,
         S390x = 5,
+        Armv6 = 6,
     }
     public enum CharSet
     {
```

### API Usage

```C#
               case Architecture.Armv6:
                   Assert.Equal(Architecture.Armv6, processArch);
                   break;
```

**Comment by terrajobst** [comment]
* Looks good as proposed

```C#
namespace System.Runtime.InteropServices
{
    public enum Architecture
    {
        // Existing:
        // X86
        // X64
        // Arm
        // Arm64
        // Wasm
        // S390x
        Armv6,
    }
}
```

**Comment by huoyaoyuan** [comment]
Implemented in #62594
